documentation/docs/14-content-security-policy.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Content Security Policy
+---
+
+At the moment, SvelteKit supports adding Content Security Policy via hooks. In environments with a runtime, HTTP headers can be added to the response object.
+
+However, SvelteKit also requires some small pieces of inline JavaScript in order for hydration to work. To avoid using `'unsafe-inline'` (which, as the name suggests, should be avoided), SvelteKit can be configured to inject CSP nonces into the HTML it generates.
+
+The nonce value is availiable to hooks as `request.locals.nonce`. A basic CSP handler hook might then look like this:
+
+```javascript
+export async function handle ({ request, resolve }) {
+  const directives = {
```

Comment:
can you move this down to line 26 where it's first used? you could also then add the nonces where you declare this instead of calling `.push`